cg/meta/workflow/rnafusion.py

```diff
@@ -134,23 +131,14 @@ def config_case(
         self.write_params_file(case_id=case_id, pipeline_parameters=pipeline_parameters.dict())
 
     def report_deliver(self, case_id: str) -> None:
-        """Get a deliverables file template from resources, parse it and, then write the deliverables file."""
-        deliverables_content: dict = self.get_template_deliverables_file_content(
-            resources.RNAFUSION_BUNDLE_FILENAMES_PATH
+        """Create deliverables file."""
+        deliverables_content: RnafusionDeliverables = (
+            RnafusionDeliverables.get_deliverables_for_case(
```

Comment:
Using a class function as a parameter for class instantiation seems weird to me.

Reply:
See a possible solution to this below

Reply:
I think the design looks nice actually! My understanding is that you have the blueprint BaseModel PipelineDeliverables, which RnafusionDeliverables implement. Then get_deliverables_for_case when called in RnafusionDeliverables formats the files found for that case in the way needed in the Rnafusion pipeline. 

Also, I think if you do not use the class method to instantiate,  then you would need to provide the files already unless you modify the parent class.

Although I feel that you could potentially consider only passing case_id to get_deliverables_for_case if the refactoring needed to move the get_class_path method is small.

Reply:
I was thinking that the function should not return the list of files but assign them to `RnafusionDeliverables.files` instead

Reply:
Oh, but what would be RnafusionDeliverables object in that case? Because I don't think you can instantiate that class without having <files> from the init

Reply:
I would say modify also the PipelineDelivery class so it doesn't need `files` to be instantiated.

Reply:
Fair enough. I think both approaches can be valid but I don't have any issues with how it is written now, though.

Reply:
this class has been removed. see comment above